src/tscore/X509HostnameValidator.cc

```diff
@@ -166,7 +166,7 @@ wildcard_match(const unsigned char *prefix, size_t prefix_len, const unsigned ch
    * permitted characters and only matches a single label
    */
   for (p = wildcard_start; p != wildcard_end; ++p) {
-    if (!(('0' <= *p && *p <= '9') || ('A' <= *p && *p <= 'Z') || ('a' <= *p && *p <= 'z') || *p == '-')) {
+    if (!(('0' <= *p && *p <= '9') || ('A' <= *p && *p <= 'Z') || ('a' <= *p && *p <= 'z') || *p == '-' || *p == '_')) {
```

Comment:
This should be ordered to have the most likely match first.  I assume it should be lowercase, digit, uppercase, and then symbols.